core/src/main/java/io/questdb/cairo/TableWriter.java

```diff
@@ -2313,44 +2330,32 @@ public Row newRow() {
     public Row newRow(long timestamp) {
         switch (rowAction) {
             case ROW_ACTION_NO_PARTITION:
-
-                if (timestamp < Timestamps.O3_MIN_TS) {
-                    throw CairoException.nonCritical().put("timestamp before 1970-01-01 is not allowed");
-                }
-
+                validateDesignatedTimestampBounds(timestamp);
                 if (timestamp < txWriter.getMaxTimestamp()) {
-                    throw CairoException.nonCritical().put("cannot insert rows out of order to non-partitioned table. Table=").put(path);
+                    throw CairoException.nonCritical()
+                            .put("cannot insert rows out of order to non-partitioned table. Table=").put(path);
                 }
-
                 bumpMasterRef();
                 updateMaxTimestamp(timestamp);
                 break;
             case ROW_ACTION_NO_TIMESTAMP:
                 bumpMasterRef();
                 break;
             case ROW_ACTION_O3:
+                validatePartitioningTimestampBounds(timestamp);
```

Comment:
why can't we validate timestamp on entry to this method?

Reply:
Same reasoning as above, I wanted to first distinguish whether the timestamp is used for partitioning or not.